src/main/scala/java/time/Instant.scala

```diff
@@ -0,0 +1,267 @@
+package java.time
+
+import scala.scalajs.js
+
+import java.time.temporal._
+
+/** Created by alonsodomin on 26/12/2015. */
+final class Instant private (seconds: Long, nanos: Int)
+    extends TemporalAccessor with Temporal with TemporalAdjuster
+    with Comparable[Instant] with java.io.Serializable {
+
+  import Preconditions._
+  import Constants._
+  import Instant._
+  import ChronoField._
+  import ChronoUnit._
+
+  requireDateTime(seconds >= MinSecond && seconds <= MaxSecond,
+      s"Invalid seconds: $seconds")
+  requireDateTime(nanos >= 0 && nanos <= MaxNanosInSecond,
+      s"Invalid nanos: $nanos")
+  if (seconds == MaxSecond && nanos > 0)
+    throw new DateTimeException(s"Specified instant is bigger than maximum")
+
+  private val totalNanos = nanos + (seconds * NANOS_IN_SECOND.toLong)
```

Comment:
Consider keeping `totalNanos` as the only filed in `Instant` because `seconds` and `nanos` are redundant.